sorc/rtma3d_read_diag.fd/rtma3d_read_diag.f90

```diff
@@ -0,0 +1,285 @@
+  namelist/iosetup/ infilename, outfilename, dump_pseudo_obs_too 
+!
+! output variables
+!
+  character(len=3)  :: var
+  real(r_single) :: rlat,rlon,rprs,rhgt,robs1,rdpt1,robs2,rdpt2,ruse,rerr
+  real(r_single) :: rdhr,iusev,ddiff
+  logical        :: l_tdry          ! if tdry is available in obsdiag for qob only
+  real(r_single) :: rtdry           ! dry-bulb air temp (q obs only)
+  character(8) :: stationID
+  integer(i_kind) :: itype,iuse
+  integer(i_kind) :: isubtype,isubtype0
+!
+!  misc.
+!
+  character ::  ch
+  integer(i_kind) :: i,j,k,ios
+  integer(i_kind) :: ic, iflg
+
+  logical fexist
+
+!
+  data infilename /'diag_conv.dat'/
+  data outfilename /'diag_results'/
+  data dump_pseudo_obs_too /.false./
+
+  inquire(file='namelist.conv',exist=fexist)
+  if(fexist) then
+     open(11,file='namelist.conv')
+      read(11,iosetup)
+     close(11)
+  endif
+!
+  open(42, file=trim(outfilename),IOSTAT=ios)
+  if(ios > 0 ) then
+       write(*,*) ' cannot open file ', trim(outfilename)
+       stop 123
+  else
+       write(*,*) ' open file ', trim(outfilename)
+  endif
+!
+  OPEN (17,FILE=trim(infilename),STATUS='OLD',IOSTAT=ios,ACCESS='SEQUENTIAL',  &
+             FORM='UNFORMATTED')
+     if(ios > 0 ) then
+       write(*,*) ' file is unavailabe: ', trim(infilename)
+       stop 123
+     endif
+
+     read(17, ERR=999) idate
+     write(*,*) 'process date: ',idate
+100  continue
+!     read(17, ERR=999,end=110) var, nchar,nreal,ii,mype
+!     write(*,*) var, nchar,nreal,ii,mype
+     read(17, ERR=999,iostat=ios,end=110) var, nchar,nreal,ii,mype,ioff0
+     write(*,*) var, nchar,nreal,ii,mype,ioff0
+     l_tdry = .false.                                 ! reset l_tdry to false (as default for all obs)
+     if ( var(1:3) == '  q' .and. ioff0 >= 22) l_tdry = .true.
+     if (ii > 0) then
+          allocate(cdiagbuf(ii),rdiagbuf(nreal,ii),cprvstg(ii),csprvstg(ii))
+          read(17,ERR=999,end=110) cdiagbuf, rdiagbuf
+          cprvstg='XXXXXXXX' 
+          csprvstg='XXXXXXXX'
+          if (var(1:3)=='  t' .or. var(1:3)=='  q' .or. var(2:3)=='ps' .or. &
+              var(2:3)=='uv'  .or. var(1:3)=='spd' .or. var(1:3)=='gst' .or. var(1:3)=='hwv') read(17)cprvstg,csprvstg
+          do i=1,ii
+             cprovider=cprvstg(i)
+             csubprovider=csprvstg(i)
+             itype=rdiagbuf(1,i)    ! observation type
+             isubtype=rdiagbuf(2,i) ! observation subtype
+             rlat=rdiagbuf(3,i)     ! observation latitude (degrees)
+             rlon=rdiagbuf(4,i)     ! observation longitude (degrees)
+             rprs=rdiagbuf(6,i)     ! observation pressure (hPa)
+             rhgt=rdiagbuf(7,i)     ! observation height (meters)
+             rdhr=rdiagbuf(8,i)     ! obs time (hours relative to analysis time)
+             iuse=int(rdiagbuf(12,i))    ! analysis usage flag (1=use, -1=monitoring ) 
+             iusev=int(rdiagbuf(11,i))    ! analysis usage flag ( value ) 
```

Comment:
@MatthewMorris-NOAA 

iusev is a real/float type variable, function "int()" should not be used here. remove the INT function and change it to
isuev=rdiagbuf(11,i)

Reply:
@GangZhao-NOAA I just committed this change to my fork on Github, so I believe it's now reflected in the PR. Can you confirm?

Reply:
@MatthewMorris-NOAA Yes, I see that you changed the code and made the commit.
Thank you!